akka-actor/src/main/scala/akka/actor/ActorRef.scala

```diff
@@ -599,6 +599,22 @@ private[akka] class VirtualPathContainer(
 
   private val children = new ConcurrentHashMap[String, InternalActorRef]
 
+  override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = message match {
+    case sel @ ActorSelectionMessage(msg, elements, _) ⇒ {
+      elements.head match {
+        case SelectChildName(name) ⇒
+          val child = getChild(name)
+          if (elements.tail.isEmpty) {
+            child ! msg
+          } else {
+            child ! sel.copy(elements = elements.tail)
+          }
+        case _ ⇒ // .. and * are unsupported
```

Comment:
In no case do we have an explanation of how the message ended up in deadLetters, if we add one then we need to do that everywhere. I’m not saying that that is a bad idea, but it sure is not something to be done in this PR. We should open a new ticket about that part and use EmptyLocalActorRef here as I suggested. Then, if we enhance ELAR to include a reason we can add an appropriate one in this instance, too.